### ISSUE TITLE
Java container: Could not find any process

### ISSUE THREAD

**Issue by sbaier1** [VerizonMedia/kubectl-flame#14]
I'm trying to profile a Java container (`hivemq/hivemq4:k8s-4.4.1` to be specific, deployed using [this chart](https://github.com/hivemq/helm-charts/tree/master/charts/hivemq-operator)), however i am seeing the above `Error: could not find any process` error.

My Kubernetes version is `1.18.8`, deployed with a pretty standard kubespray inventory.

The process is just regularly running as PID 1 (and can also be found in `/proc/1/`) and i have also used async profiler manually from within the container before so there should be no issues. Maybe the process name search isn't working here?

**Comment by edeNFed** [comment]
Hi, @sbaier1 thanks for opening this issue. 
Can you please verify if this issue still exists after PR #15?
You can easily upgrade to the latest kubectl-flame version via krew by running: `kubectl krew upgrade`
Thanks

**Comment by sbaier1** [comment]
Hey @edeNFed , thanks for the follow-up! I am indeed not seeing this specific error anymore now, however the tool still isn't working as expected for me, but this time the behavior is a bit more bizarre:
When profiling i am seeing this error: `Profiling ... Could not open /tmp/async-profiler/flamegraph.svg`

and i am also seeing restarts of the container i tried to profile. so i checked the logs at the time of profiling, and somehow it seems that the profiling agent crashes the JVM:

```
Profiling ... Could not open /tmp/async-profiler/flamegraph.svg
Error: open /tmp/async-profiler/flamegraph.svg: no such file or directory
Could not open /tmp/async-profiler/flamegraph.svg



#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000000000ea16, pid=1, tid=113
#
# JRE version: OpenJDK Runtime Environment 18.9 (11.0.8+10) (build 11.0.8+10)
# Java VM: OpenJDK 64-Bit Server VM 18.9 (11.0.8+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  0x000000000000ea16
#
# Core dump will be written. Default location: /opt/hivemq-4.4.1/core
#
# An error report file with more information is saved as:
# /opt/hivemq-4.4.1/hs_err_pid1.log
Compiled method (c1)  166716 7288   !   3       io.netty.channel.unix.NativeInetAddress::address (134 bytes)
 total in heap  [0x00007fceac8e6110,0x00007fceac8e77a0] = 5776
 relocation     [0x00007fceac8e6288,0x00007fceac8e63c0] = 312
 main code      [0x00007fceac8e63c0,0x00007fceac8e6e80] = 2752
 stub code      [0x00007fceac8e6e80,0x00007fceac8e6f50] = 208
 oops           [0x00007fceac8e6f50,0x00007fceac8e6f58] = 8
 metadata       [0x00007fceac8e6f58,0x00007fceac8e6fa0] = 72
 scopes data    [0x00007fceac8e6fa0,0x00007fceac8e7290] = 752
 scopes pcs     [0x00007fceac8e7290,0x00007fceac8e7620] = 912
 dependencies   [0x00007fceac8e7620,0x00007fceac8e7638] = 24
 handler table  [0x00007fceac8e7638,0x00007fceac8e77a0] = 360
Compiled method (c1)  166716 7288   !   3       io.netty.channel.unix.NativeInetAddress::address (134 bytes)
 total in heap  [0x00007fceac8e6110,0x00007fceac8e77a0] = 5776
 relocation     [0x00007fceac8e6288,0x00007fceac8e63c0] = 312
 main code      [0x00007fceac8e63c0,0x00007fceac8e6e80] = 2752
 stub code      [0x00007fceac8e6e80,0x00007fceac8e6f50] = 208
 oops           [0x00007fceac8e6f50,0x00007fceac8e6f58] = 8
 metadata       [0x00007fceac8e6f58,0x00007fceac8e6fa0] = 72
 scopes data    [0x00007fceac8e6fa0,0x00007fceac8e7290] = 752
 scopes pcs     [0x00007fceac8e7290,0x00007fceac8e7620] = 912
 dependencies   [0x00007fceac8e7620,0x00007fceac8e7638] = 24
 handler table  [0x00007fceac8e7638,0x00007fceac8e77a0] = 360
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#

[error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007fcec713d611]

```

should i raise a separate issue for this?

**Comment by edeNFed** [comment]
Hi @sbaier1 after further investigation I think the problem you are seeing is that the target java process can not access the location created by the profiler. I fixed it in #18 
After this fix I am now able to profile the following deployment successfully on my EKS cluster:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: hivemq
  name: hivemq
  namespace: eden
spec:
  template:
    metadata:
      labels:
        app: hivemq
    spec:
      containers:
      - image: hivemq/hivemq4:k8s-4.4.1
        imagePullPolicy: IfNotPresent
        name: hivemq
        ports:
        - containerPort: 8080
          protocol: TCP
```
Output:
```
➜ kubectl flame hivemq-78f89848cf-dnftf --lang java -t 20s -f flame.svg
Verifying target pod ... ✔
Launching profiler ... ✔
Profiling ... ✔
FlameGraph saved to: flame.svg 🔥
```

Please upgrade and let me know if this fix is working for you.

**Comment by sbaier1** [comment]
nice, thanks for fixing this 👍 
it's working perfectly now.
